task/distributed_queue.py

```diff
@@ -105,14 +106,15 @@ def __init__(
                                                ssl_options=pika.SSLOptions(context, server_hostname=host),
                                                credentials=pika.credentials.ExternalCredentials())
 
+        self.address = amqp_server
         self.connection = pika.BlockingConnection(params)
         self.channel = self.connection.channel()
         self.queue_counts: dict[str, int] = {}
 
         for queue in queues:
             try:
                 result = self.channel.queue_declare(queue=queue, arguments=arguments.get(queue, None), **kwargs)
-                self.queue_counts[queue] = result.method.message_count
+                self.queue_counts[queue] = result.method.message_count or 0
```

Comment:
I that `or 0` necessary? It wasn't before, and it really shouldn't be `None`. If so, this requires an explanation (possibly in a separate commit?)

Reply:
So according to the spec:
 - https://www.rabbitmq.com/amqp-0-9-1-reference#queue.declare-ok.message-count
 - https://www.rabbitmq.com/amqp-0-9-1-reference#domain.message-count

There is no mention that this might be nullable.

Also, looking at the code:

```python
    class DeclareOk(amqp_object.Method):
        ...
        def decode(self, encoded, offset=0):
            ...
            self.message_count = struct.unpack_from('>I', encoded, offset)[0]
            ...
```

Which is to say, that no `DeclareOk` created from calling `.decode()` is ever going to have this field as `None`, but if `.decode()` hasn't been called, it might be.  This looks like a case of bad binding...

I feel like an assert is probably more appropriate here, indeed.